spectator-ext-aws/src/main/java/com/netflix/spectator/aws/SpectatorRequestMetricCollector.java

```diff
@@ -72,37 +72,47 @@
   };
 
   private final Registry registry;
+  private final Map<String, String> customTags;
 
   /**
    * Constructs a new instance.
    */
   public SpectatorRequestMetricCollector(Registry registry) {
-    super();
+    this(registry, Collections.emptyMap());
+  }
+
+  /**
+   * Constructs a new instance. Custom tags provided by the user will be applied to every metric.
+   * If there is a conflict between a tag added by SpectatorRequestMetricCollector and a custom tag,
+   * the custom tag will override the built-in one.
+   */
+  public SpectatorRequestMetricCollector(Registry registry, Map<String, String> customTags) {
     this.registry = Preconditions.checkNotNull(registry, "registry");
+    this.customTags = Preconditions.checkNotNull(customTags, "customTags");
```

Comment:
A couple other options:
- provide a setter, so that a user can change the tags sent after construction
- a protected method that can be overridden by subclassing (eg making getAllTags protected to it can be overridden), which would allow users to specify custom logic to tag their metrics with
The latter option is most powerful - thoughts?

Reply:
The map would need to get copied to ensure it cannot be mutated after being passed in.

Either option you list there would have additional risks of thread safety issues.

Reply:
Ah right, thanks.